.github/workflows/deploy-catalog.yaml

```diff
@@ -0,0 +1,53 @@
+name: Deploy catalog to ECR
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - '.github/workflows/deploy-catalog.yaml'
+      - 'catalog/**'
+      - 'shared/**'
+
+jobs:
+  deploy-catalog-ecr:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: catalog
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '16.11'
+          cache: 'npm'
```

Comment:
But does this do anything by itself, without the `cache-dependency-path`? From the docs, it sounds like it will only look for `package-lock.json` in the root directory, and not in `catalog/package-lock.json`.

Reply:
> But does this do anything by itself, without the cache-dependency-path? 

i assumed that it would just never invalidate the cache, but that turned out wrong. i fixed it by bringing back `catalog/package-lock.json`